controller/composite/controller.go

```diff
@@ -471,7 +471,7 @@ func (pc *parentController) makeSelector(parent *unstructured.Unstructured, extr
 }
 
 func (pc *parentController) canAdoptFunc(parent *unstructured.Unstructured) func() error {
-	nsParentClient := pc.parentClient.WithNamespace(parent.GetNamespace())
+	nsParentClient := pc.parentClient
```

Comment:
Do we actually need to reassign this variable here? Could we just directly call `ps.parentClient.Get(...)` below?

Reply:
Agreed.

Reply:
I agree too. In addition, make sure `pc.parentClient` is a namespace-agnostic client since a single `parentController` instance manages objects in multiple namespaces. In other words, the line below should ultimately read `pc.parentClient.Namespace(parent.GetNamespace()).Get(parent.GetName(), ...)`.

Reply:
Noted, thanks for pointing this out. :)